cpp/test/sg/fnv_hash_test.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/fil/fnv_hash.h>
+#include <gtest/gtest.h>
+#include <raft/error.hpp>
+
+typedef std::tuple<std::vector<char>, unsigned long long, unsigned long>
+  fnv_vec_t;
+
+class FNVHashTest : public testing::TestWithParam<fnv_vec_t> {
+ protected:
+  void SetUp() override { std::tie(input, correct64, correct32) = GetParam(); }
+
+  void check() {
+    unsigned long long real64 =
```

Comment:
`real32` and `real64` remind of floating-point types. `computed`, `got` or `hash` would be better.